### PR TITLE
Change defaults for BaseSettings

### PR DESCRIPTION
**Pull request by dmontagu**
## Change Summary

Sets the defaults for BaseSettings in line with #576.

Just trying to chip away at the v1.0 milestone.

## Related issue number

#721

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.rst` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Comment by dmontagu** [comment]
@samuelcolvin I ran into a surprise while ensuring case-sensitivity was indeed being checked: https://stackoverflow.com/questions/19023238/why-python-uppercases-all-environment-variables-in-windows

It looks like:
1. There wasn't actually a test that case-sensitivity was required if the setting was enabled
1. On Windows, the environment is read in a case-insensitive way.

Because of the second point above, I'm not sure there is a great way to even test this on the build server. I've added a test using the pytest `monkeypatch` fixture to ensure case sensitivity to get the expected failure even on Windows, but let me know if you would prefer a different approach (or to just drop this test? Apparently it isn't required for 100% coverage...).

-----

Separately, it sounds like you can access a case-sensitive version of the environment via python's platform-specific `nt`module. That said, I personally would be inclined to just leave a note/warning in the docs about this for Windows, and not actually handle case sensitivity there. Do you agree?

I've marked it as a note in the docs now, but can change it to a warning if you prefer, just let me know.

**Comment by dmontagu** [comment]
@samuelcolvin  +1 for removing case-sensitivity for the environment variables altogether.

**Comment by samuelcolvin** [comment]
**Some how managed to post this on the wrong issue:**

Couple of points:
* We can add windows only tests, since CI runs in windows and linux that should work fine, just `skipIf`.
* I'd be entirely up for removing case sensitivity altogether, I can't imagine a situation where you'd ever **desire** case sensitivity. It was added as an option to avoid a breaking change, but if we think that switch is superfluous, now would be the time to remove it.

I've been thinking about how `BaseSettings` works, I'll add more general points to #747.

Let's agree there before we move forward here.

File: pydantic/env_settings.py [hunk 1]
```diff
@@ -57,10 +57,10 @@ def _build_environ(self) -> Dict[str, Optional[str]]:
         return d
 
     class Config:
-        env_prefix = 'APP_'
+        env_prefix = ''
         validate_all = True
         extra = Extra.forbid
         arbitrary_types_allowed = True
-        case_insensitive = False
+        case_insensitive = True
```
Comment (samuelcolvin) [inline pydantic/env_settings.py]: `case_insensitive = False` is a double negative.

Let's take this opportunity to change the name to `case_sensitive = False`.

File: pydantic/env_settings.py [hunk 2]
```diff
@@ -13,9 +13,6 @@ class BaseSettings(BaseModel):
     """
     Base class for settings, allowing values to be overridden by environment variables.
 
-    By default environment variables must be upper case and prefixed by APP_ by default. Eg. to override foobar,
-    `export APP_FOOBAR="whatever"`. To change this behaviour set Config options case_insensitive and env_prefix.
-
```
Comment (dmontagu) [inline pydantic/env_settings.py]: @samuelcolvin I wasn't sure if I should include a modified description here -- the defaults seem a little more intuitive now, so maybe this just isn't necessary?

Comment (samuelcolvin) [reply pydantic/env_settings.py]: makes sense.

**Pull request merged**